tests/suites/test_suite_psa_crypto_entropy.data

```diff
@@ -1,3 +1,30 @@
+PSA external RNG failure: generate random and key
+external_rng_failure_generate:
+
+# When verifying the impact of a forced RNG failure, depend on the built-in
+# implementation of the algorithm that uses randomization, whether it's
+# because the algorithm is randomized or because our implementation uses
+# randomization for (e.g.) blinding. An external implementation could use
+# its own randomness source which is not affected by the forced failure of
+# the RNG driver.
+# Key types and non-randomized auxilary algorithms (in practice, hashes) can
+# use an external implementation.
+PSA external RNG failure: randomized ECDSA
+depends_on:PSA_WANT_KEY_TYPE_ECC_KEY_PAIR:MBEDTLS_PSA_BUILTIN_ALG_ECDSA:MBEDTLS_ECP_DP_SECP256R1_ENABLED
+external_rng_failure_sign:PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):"ab45435712649cb30bbddac49197eebf2740ffc7f874d9244c3460f54f322d3a":PSA_ALG_ECDSA_ANY:32
+
+PSA external RNG failure: deterministic ECDSA (software implementation)
+depends_on:PSA_WANT_KEY_TYPE_ECC_KEY_PAIR:MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA:MBEDTLS_ECP_DP_SECP256R1_ENABLED:MBEDTLS_SHA256_C:PSA_WANT_ALG_SHA_256
```

Comment:
Currently yes. It's a [limitation with compound algorithms](https://github.com/ARMmbed/mbedtls/issues/3997). (That such cases might not work out of the box was a deliberate simplification in the conditional inclusion mechanism, so that it could be designed and implemented quickly.) Deterministic ECDSA uses SHA-256 under the hood, and the builtin implementation calls `mbedtls_sha256_xxx`. Since deterministic ECDSA uses SHA-256 internally, it needs both `MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA` and `MBEDTLS_SHA256_C` (and for the same reason it needs `MBEDTLS_ECP_DP_SECP256R1_ENABLED` and not `PSA_WANT_ECC_FAMILY_SECP_R1`). In addition, hashing the message to get the hash that is signed with `psa_sign_hash`, as well as potentially metadata management in `psa_sign_hash`, needs `PSA_WANT_ALG_SHA_256`.

Reply:
This is a bit of a dark corner in the design, so until this is resolved, I think test dependencies can have whatever works in practice. It's only a problem for the rare tests that require the built-in implementation.

Reply:
ok, that makes sense to me, thanks for the explanation.